src/Microsoft.OData.Client/DataServiceClientFormat.cs

```diff
@@ -109,6 +114,40 @@ private set
         }
 
         /// <summary>
+        /// Gets the flag for using Json in current batch request.
+        /// Default value is false.
+        /// For now, ODL-client sets both the Accept and Content-Type headers to application/json
+        /// if this flag is set to true.
+        /// </summary>
+        internal bool IsJsonBatchFormat { get; private set; }
+
+        /// <summary>
+        /// Indicates that the client should use the JSON format for the batch request.
+        /// Will invoke the LoadServiceModel delegate property in order to get the required service model.
+        /// </summary>
+        public void UseJsonForBatch()
+        {
+            if (this.ServiceModel == null)
+            {
+                throw new InvalidOperationException(Strings.DataServiceClientFormat_LoadServiceModelRequired);
+            }
+
+            this.IsJsonBatchFormat = true;
+        }
+
+        /// <summary>
+        /// Indicates that the client should use the JSON format for the batch request.
+        /// Will invoke the LoadServiceModel delegate property in order to get the required service model.
+        /// </summary>
+        public void UseJsonForBatch(IEdmModel serviceModel)
+        {
+            Util.CheckArgumentNull(serviceModel, "serviceModel");
+
+            this.IsJsonBatchFormat = true;

```

Comment:
>this.IsJsonBatchFormat = true; [](start = 12, length = 30)

can you remove this and call the other overload after setting the service model?